misk-hibernate-testing/src/main/kotlin/misk/jdbc/StartVitessService.kt

```diff
@@ -108,6 +162,70 @@ internal class DockerVitessCluster(
         .exec(LogContainerResultCallback())
         .awaitStarted()
     StartVitessService.logger.info("Started Vitess with container id $containerId")
+
+    waitUntilHealthy()
+
+    grantExternalAccessToDbaUser()
+
+    turnOnGeneralLog()
+  }
+
+  private fun waitUntilHealthy() {
+    retry(10, ExponentialBackoff(Duration.ofMillis(20), Duration.ofMillis(1000))) {
+      cluster.connect().use { c ->
+        try {
+          val result =
+              c.createStatement().executeQuery("SELECT 1 FROM dual").uniqueResult { it.getInt(1) }
+          check(result == 1)
+        } catch (e: Exception) {
+          val message = e.message
+          if (message != null && message.contains("table dual not found")) {
+            throw RuntimeException(
+                "Something is wrong with your vschema and unfortunately vtcombo does not " +
+                    "currently have good error reporting on this. Please do an ocular inspection.")
```

Comment:
“Please do an ocular inspection” is cute but it also jargony! Like, “is ocular a component of Vitess?”

Reply:
Haha fair enough, reworded.